dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java

```diff
@@ -989,21 +989,21 @@ private DynamicConfiguration getDynamicConfiguration(URL connectionURL) {
      */
     private final AtomicInteger serviceRefreshState = new AtomicInteger(0);
 
-    private void registerServiceInstance() {
-        try {
-            registered = true;
+    public synchronized void registerServiceInstance() {
```

Comment:
`checkState` already uses synchronized, so consider deleting it here.

Reply:
This might be called by `ServiceConfig` and `ServiceConfig` can be parallel.